server.go

```diff
@@ -1500,6 +1506,13 @@ func (s *Server) ReadSeries(database, retentionPolicy, name string, tags map[str
 // Returns a resultset for each statement in the query.
 // Stops on first execution error that occurs.
 func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) Results {
+	// Authorize user to execute the query.
+	if err := Authorize(user, q, database); err != nil {
```

Comment:
Authorization should happen entirely before anything gets executed. Just like if one of the statements doesn't parse. If we can know it up front, we should check it and send an error without getting into some half state.